### DISCUSSION TITLE
Sligthly better anonymous givens names

### DISCUSSION THREAD

**Discussion by Lasering** [scala/scala3#19179] [Feature Requests]
The following:
```scala
object Connector:
  enum Type:
    case CHAdeMO, Mennekes

object Parking:
  enum Type:
    case AlongMotorway, OnStreet

trait Decoder[T]

given Decoder[Connector.Type] = new Decoder[Connector.Type] {}
given Decoder[Parking.Type] = new Decoder[Parking.Type] {}
```
Fails with:
```scala
Double definition:
final lazy given val given_Decoder_Type:
  Playground.Decoder[Playground.Connector.Type] in object Playground at line 13 and
final lazy given val given_Decoder_Type:
  Playground.Decoder[Playground.Parking.Type] in object Playground at line 14
```
[Scastie link](https://scastie.scala-lang.org/VfH2g926Q92JaIaPMPlpKA)

Which is as expected reading the documentation [Anonymous Given Instances](https://docs.scala-lang.org/scala3/reference/contextual/relationship-implicits.html#anonymous-given-instances-1). Namely the point 2:
> 2) the simple name(s) of the implemented type(s), leaving out any prefixes

However this could be improved. If the user declared the given with a prefix, then use that prefix to generate the anonymous given name. So in this case the generated names would be: `given_Decoder_Connector_Type` and `given_Decoder_Parking_Type`

**Comment by sjrd** [comment]
We cannot change the generated names in a compatible way. So I'm afraid we're stuck with the current spec.

**Reply by Lasering** [reply]
It was worth a shot.

**Comment by som-snytt** [comment]
https://github.com/lampepfl/dotty/issues/18034

https://github.com/lampepfl/dotty/issues/15316

where it's recommended to just name them. I proposed auto-suggesting a unique name, as when registering for a web site with a user name or generating a password. 

Alternatively, disambiguate during incremental compilation only, with the expectation that the final edit will be unambiguous. It would warn until then.
